lib/commands/push.ts

```diff
@@ -301,9 +310,43 @@ export default class PushCmd extends Command {
 					opts,
 				};
 				await remote.startRemoteBuild(args);
+				// Maybe ^ this should return the id of the release?
+				// instead of getting the latest from below
+				if (options['tag']) {
+					const latestRelease = await sdk.models.release.getLatestByApplication(
```

Comment:
Like you hinted in the code comment above, I would also feel more comfortable if `startRemoteBuild` returned the release ID. The worry is that `release .getLatestByApplication` might return the wrong release. If the could builder is busy, it could mean minutes (?) between the time when the builder creates a new release, and the time when the CLI executes this line. Consider also, for example, two end users working on the same application, and running `balena push` more or less at the same time, and the jobs falling on different cloud builders (a fast native ARM builder, and a slow emulated build). This could mean that the build that started first could end _after_ another build submitted shortly after the first.

Reply:
And I see that you had already considered this in Flowdock too :-)
> https://www.flowdock.com/app/rulemotion/i-cli/threads/XCyAW9_JBlef1rQJdGi21gEpPbX
> The issue that I see with this approach, is that many users try to push in the same App so multiple releases are created so I am afraid that we might end up tagging the wrong release.

Reply:
We are in agreement then, I'll make the `startRemoteBuild` to return an id.